### DISCUSSION TITLE
How to achieve/measure a temperature step in a material? (Material ignition?)

### DISCUSSION THREAD

**Discussion by Elephant01XLY** [firemodels/fds#12971] [Q&A] [answered]
Hello everyone,
I want to simulate the thermal ignition of a solid material, i.e. after reaching a certain temperature, the temperature of the material will suddenly increase until it burns out. Similar to an explosion.
I have set up the thermal decomposition with reference to the FDS manual, but I can't measure the sudden increase in temperature after ignition of the object.
I don't know if it's a problem with the material settings or the temperature measurement point settings.
The code is as below, please help me to see what is wrong.

&HEAD CHID='28'/

&TIME T_END=800.0/

&DUMP DT_RESTART=300.0, DT_SL3D=0.25/

&MESH ID='Mesh01', IJK=40,40,40, XB=-1.0,1.0,-1.0,1.0,0.0,2.0/

&SPEC ID='PROPANE'/

&REAC ID='FDS6 METHANE',

      FYI='FDS6 Predefined',

      FUEL='METHANE'/

&DEVC ID='SOLID101', QUANTITY='ADIABATIC SURFACE TEMPERATURE', XYZ=0.0,0.0,0.7, IOR=3/

&DEVC ID='SOLID10', DEPTH=0.4, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.3, IOR=-3/

&DEVC ID='SOLID11', QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.3, IOR=-3/

&DEVC ID='SOLID2', DEPTH=0.399, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=-0.199,-0.199,0.3, IOR=-3/

&DEVC ID='SOLID6', DEPTH=1.0E-3, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=-0.199,-0.199,0.3, IOR=-3/

&DEVC ID='SOLID3', DEPTH=0.399, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.199,-0.199,0.3, IOR=-3/

&DEVC ID='SOLID7', DEPTH=1.0E-3, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.199,-0.199,0.3, IOR=-3/

&DEVC ID='SOLID8', DEPTH=1.0E-3, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.199,0.199,0.3, IOR=-3/

&DEVC ID='SOLID4', DEPTH=0.399, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.199,0.199,0.3, IOR=-3/

&DEVC ID='SOLID9', DEPTH=1.0E-3, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=-0.199,0.199,0.3, IOR=-3/

&DEVC ID='SOLID5', DEPTH=0.399, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=-0.199,0.199,0.3, IOR=-3/

&DEVC ID='SOLID111', QUANTITY='ADIABATIC SURFACE TEMPERATURE', XYZ=0.0,0.0,0.3, IOR=-3/

&MATL ID='1',

      SPECIFIC_HEAT=1.176,

      CONDUCTIVITY=0.4644,

      DENSITY=1845.0,

      HEAT_OF_COMBUSTION=4.78E+12,

      N_REACTIONS=1,

      SPEC_ID(1,1)='PROPANE',

      NU_SPEC(1,1)=1.0,

      A=4780.0,

      E=1.439E+5/

&SURF ID='1',

      RGB=146,130,148,

      TEXTURE_MAP='psm_carpet_grey.jpg',

      TEXTURE_WIDTH=0.6096,

      TEXTURE_HEIGHT=0.6096,

      BURN_AWAY=.TRUE.,

      MATL_ID(1,1)='1',

      MATL_MASS_FRACTION(1,1)=1.0,

      THICKNESS(1)=0.4/

&SURF ID='burner',

      COLOR='RED',

      HRRPUA=1000.0,

      TMP_FRONT=300.0/

&OBST ID='Obstruction', XB=-0.2,0.2,-0.2,0.2,0.3,0.7, SURF_ID='1'/ 

&VENT ID='Mesh Vent: Mesh01 [XMAX]', SURF_ID='OPEN', XB=1.0,1.0,-1.0,1.0,0.0,2.0/ 

&VENT ID='Mesh Vent: Mesh01 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,1.0,0.0,2.0/ 

&VENT ID='Mesh Vent: Mesh01 [YMAX]', SURF_ID='OPEN', XB=-1.0,1.0,1.0,1.0,0.0,2.0/ 

&VENT ID='Mesh Vent: Mesh01 [YMIN]', SURF_ID='OPEN', XB=-1.0,1.0,-1.0,-1.0,0.0,2.0/ 

&VENT ID='Mesh Vent: Mesh01 [ZMAX]', SURF_ID='OPEN', XB=-1.0,1.0,-1.0,1.0,2.0,2.0/ 

&VENT ID='Vent01', SURF_ID='burner', XB=-0.5,0.5,-0.5,0.5,1.0E-3,1.0E-3/ 

&BNDF QUANTITY='ADIABATIC SURFACE TEMPERATURE'/

&BNDF QUANTITY='BACK WALL TEMPERATURE'/

&BNDF QUANTITY='GAS TEMPERATURE'/

&BNDF QUANTITY='NET HEAT FLUX'/

&BNDF QUANTITY='WALL TEMPERATURE'/

&SLCF QUANTITY='TEMPERATURE', ID='1', PBX=0.0/

&SLCF QUANTITY='TEMPERATURE', ID='2', PBY=0.0/

&TAIL /

**Comment by drjfloyd** [comment]
Your combustion REAC uses METHANE, but you have defined your MATL to release PROPANE.

**Reply by Elephant01XLY** [reply]
Oh, thanks a lot. I will fix this problem !

**Reply by Elephant01XLY** [reply]
Hi! I have changed the FUEL from METHANE to PROPANE. However, the calculation stop with "ERROR: Numerical Instability - FDS stopped". Can u help me with that? Thanks a lot.

**Reply by drjfloyd** [reply]
Double check the values for all your MATL inputs.

**Comment by Elephant01XLY** [comment]
Hi, my friend! As u say, the values of my materials are wrong. And I have corrected them as below. However, I still can't get a temperature jump. In my mind, when explosives are thermally decomposed, they are violently exothermic, which leads to a sharp increase in temperature.
Can u give me some advice?

&MATL ID='1',
      SPECIFIC_HEAT=1.02,
      CONDUCTIVITY=0.302,
      DENSITY=1845.0,
      HEAT_OF_COMBUSTION=4780.0,
      N_REACTIONS=1,
      SPEC_ID(1,1)='PROPANE',
      NU_SPEC(1,1)=1.0,
      A=4.78E+12,
      E=1.439E+5/

**Reply by drjfloyd** [reply]
Modern day high explosives decompose explosively when there is a shock wave moving through the material. They require a blasting cap or other initiator to get an explosion. If you just set them on fire, they just burn.  Your cube is already immersed in a fire. It is already seeing the heat flux from a flame. The fuel you are adding from the cube isn't going to drastically change the heat flux to the cube as it isn't going to burn until it mixes with oxygen.

**Reply by Elephant01XLY** [reply]
Thank you very much! So is there a possibility that in a mesh the material is heated and suddenly gives off a lot of heat, which then leads to an extreme rise in temperature?

**Reply by drjfloyd** [reply]
Your MATL reaction does not describe the physics of a high explosive undergoing detonation where chemical bonds are being disrupted by shock in addition to heat.

**Reply by Elephant01XLY** [reply]
Is there a way to achieve this in FDS? For example, measuring the temperature profile shown below
![Uploading 1.JPG…]()

**Comment by Elephant01XLY** [comment]
sorry for the preview picture, I have attached a new one. Do you think it is possible to measure the temperature profile in FDS as shown? It should be the curve of an explosive igniting and decomposing, with a violent exotherm.

![1](https://github.com/firemodels/fds/assets/155599458/7750fcd4-4874-4cf2-bb7c-256f680415db)

**Reply by drjfloyd** [reply, marked as answer]
I have no idea what exaclty is being plotted or what the source of that data is.

Explosive materials do not generate heat solely by decomposing into a gas that burns outside the explosive material. Energy is released in the decomposition process itself. In a detonation the temperature rise is not primarily due to conduction but rather pressure shock. FDS is a low Mach number code that is not going to model the pressure compoment of the decomposition.

**Reply by Elephant01XLY** [reply]
Thanks very much. Your answer solved a problem that had been bothering me for a long time。
